src/dune_engine/action_exec.ml

```diff
@@ -137,6 +138,19 @@ module Exec_result = struct
         (List.map errs ~f:(fun e -> Exn_with_backtrace.capture (Error.to_exn e)))
   ;;
 end
+type action_res = 
+  { done_or_more_deps : Action_plugin.done_or_more_deps;
+    
+    needed_deps : Dep.Set.t
+  }
```

Comment:
This type should be defined in its own submodule with a `union` function instead of `action_res_union` below.